app/DynamicsAdapter/Fams3Adapter.Dynamics/Config/DuplicateDetectionConfigService.cs

```diff
@@ -9,31 +14,86 @@ namespace Fams3Adapter.Dynamics.Config
 {
     public interface IDuplicateDetectionConfigService
     {
-        Task<IEnumerable<SSG_DuplicateDetectionConfig>> GetDuplicateDetectionConfig(CancellationToken cancellationToken);
+        Task<string> GetDuplicateDetectHashData(object entity);
     }
 
     public class DuplicateDetectionConfigService : IDuplicateDetectionConfigService
     {
         private readonly IODataClient _oDataClient;
-        public static IEnumerable<SSG_DuplicateDetectionConfig> configs;
+        public static IEnumerable<SSG_DuplicateDetectionConfig> _configs;
 
         public DuplicateDetectionConfigService(IODataClient oDataClient)
         {
             this._oDataClient = oDataClient;
         }
 
         /// <summary>
-        /// get all duplication config
+        /// make the entity hash data fields according to configuration.
+        /// Example: config is : ssg_person, ssg_firstname|ssg_lastname
+        ///     ssg_person person1: firstname="person1", lastname="lastname1"
+        ///     it should return SHA512("person1lastname1")
         /// </summary>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
-        public async Task<IEnumerable<SSG_DuplicateDetectionConfig>> GetDuplicateDetectionConfig(CancellationToken cancellationToken)
+        /// <param name="entity"></param>
+        /// <returns>SHA512 </returns>
+        public async Task<string> GetDuplicateDetectHashData(object entity)
         {
-            if (configs != null) return configs;
+            if (_configs == null) await GetDuplicateDetectionConfig(CancellationToken.None);
+
+            Type type = entity.GetType();
+            string name = type.Name switch 
+                            { 
+                                "PersonEntity" => "ssg_person",
+                                _ => ""
+                            };
```

Comment:
Can we put this in a dictionary and use the type.Name as a key to retrieve the ssg_{entity_name}

Reply:
sure